apax/nodes/selection.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import typing
+
+import ase.io
+import numpy as np
+import zntrack.utils
+from ipsuite.analysis.ensemble import plot_with_uncertainty
+from ipsuite.configuration_selection.base import BatchConfigurationSelection
+from ipsuite.utils.combine import get_flat_data_from_dict
+from matplotlib import pyplot as plt
+
+from apax.bal import kernel_selection
+from apax.nodes.model import ApaxBase
+
+log = logging.getLogger(__name__)
+
+
+class BatchKernelSelection(BatchConfigurationSelection):
+    """Interface to the batch active learning methods implemented in apax.
+    Check the apax documentation for a list and explanation of implemented properties.
+
+    Attributes
+    ----------
+    models: Union[Apax, List[Apax]]
+        One or more Apax models to construct a feature map from.
+    base_feature_map: dict
+        Name and parameters for the feature map transformation.
+    selection_method: str
+        Name of the selection method to be used. Choose from:
+        ["max_dist", ]
+    n_configurations: int
+        Number of samples to be selected.
+    processing_batch_size: int
+        Number of samples to be processed in parallel.
+        Does not affect the result, just the speed of computing features.
+    """
+
+    _module_ = "apax.nodes"
+
+    models: typing.List[ApaxBase] = zntrack.deps()
+    base_feature_map: dict = zntrack.params({"name": "ll_grad", "layer_name": "dense_2"})
+    selection_method: str = zntrack.params("max_dist")
+    n_configurations: str = zntrack.params()
+    processing_batch_size: str = zntrack.meta.Text(64)
+    img_selection = zntrack.outs_path(zntrack.nwd / "selection.png")
+
+    def select_atoms(self, atoms_lst: typing.List[ase.Atoms]) -> typing.List[int]:
+        if isinstance(self.models, list):
+            param_files = [m._parameter["data"]["directory"] for m in self.models]
+        else:
+            param_files = self.models._parameter["data"]["directory"]
+
+        if isinstance(self.train_data, dict):
+            self.train_data = get_flat_data_from_dict(self.train_data)
```

Comment:
Are we sure we need this, training_data should probably be `list[Atoms]` like usual?

Reply:
ah yes, this is something I screwed up in the batch configuration selection base class. I'll also add a PR for IPS.